File: back/dentalium/src/main/java/com/romoreno/dentalium/infraestructure/adapter/in/controller/rest/medical_study/dto/ListMedicalStudiesResponse.java
package com.romoreno.dentalium.infraestructure.adapter.in.controller.rest.medical_study.dto;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@AllArgsConstructor
public class ListMedicalStudiesResponse {

    private String patientId;
    private List<MedicalStudy> medicalStudies;

    public ListMedicalStudiesResponse() {
        medicalStudies = new ArrayList<>();
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class MedicalStudy {
        private Integer id;
        private Integer doctorId;
        private StudyType studyType;
        private Date date;
        private String filename;
    }

    @Data
    @AllArgsConstructor
    @NoArgsConstructor
    public static class StudyType {
        private Integer id;
        private String description;
    }
}

File: front/dentalium/src/app/model/api/ListMedicalStudiesResponse.ts
export class ListMedicalStudiesResponse {
  patientId: string;
  medicalStudies: MedicalStudy[];

  constructor() {
    this.patientId = '';
    this.medicalStudies = [];
  }
}

export interface MedicalStudy {
  id: number;
  doctorId: number;
  studyType: StudyType;
  date: Date;
  filename: string;
}

export interface StudyType {
  id: number;
  description: string;
}
